util/notify.go

```diff
@@ -0,0 +1,64 @@
+package util
+
+/**
+ * urfave/cli Notify "Extension"
+ *
+ * This sub-package of rig is intended for potential separation into a
+ * standalone package or PR to the main framework. As such, it should not have
+ * any built-in assumptions or identifiers about Rig, nor depend on anything in
+ * the rig codebase.
+ */
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/martinlindhe/notify"
+	"github.com/urfave/cli"
+)
+
+var config *NotifyConfig
+
+// NotifyConfig holds configuration for notification support
+type NotifyConfig struct {
+	// The label for the app to be displayed in notifications.
+	Label string
+
+	// Relative path to notification logo.
+	Icon string
+}
+
+// NotifyInit initializes notification config
+func NotifyInit(label string) error {
+	config = &NotifyConfig{
+		Icon:     "util/logo.png",
+		Label:    label,
+	}
+	return nil
+}
+
+// NotifySuccess send a notification for a successful command run
+func NotifySuccess(ctx *cli.Context, message string) {
+	if shouldNotify() {
+		notify.Notify(config.Label, fmt.Sprintf("Success: %s", ctx.Command.Name), message, config.Icon)
+	}
+}
+
+// NotifyError send a notification for a failed command run
+func NotifyError(ctx *cli.Context, message string) error {
+	if shouldNotify() {
+		notify.Notify(config.Label, fmt.Sprintf("Error: %s", ctx.Command.Name), message, config.Icon)
+	}
+	return nil
+}
+
+// shouldNotify returns a boolean if notifications are enabled
+func shouldNotify() bool {
+	env, isSet := os.LookupEnv("RIG_SILENCE_NOTIFY")
```

Comment:
My sense is that RIG_NOTIFY_SILENCE is a better name, as there is more likely to be other `RIG_NOTIFY_*` vars than `RIG_SILENCE_*`

Reply:
Should this be a global parameter, allowing `rig --silence`

Reply:
Most apps use `--quiet` but that is generally targeted at log messages, not notifications.  I can go either way on a CLI flag for it. Do you feel strongly one way or another?

(I do agree with the name change)